skymap_scanner/server/start_scan.py

```diff
@@ -163,7 +165,13 @@ def pixel_already_sent(pixel: Tuple[icetray.I3Int, icetray.I3Int]) -> bool:
         # find pixels to refine
         LOGGER.info(f"Looking for refinements for {nside_subprogression}...")
         #
-        pixels_to_refine = choose_pixels_to_reconstruct(self.nsides_dict, nside_subprogression)
+        pixels_to_refine = choose_pixels_to_reconstruct(
+            self.nsides_dict,
+            nside_subprogression,
+            ang_dist=self.ang_dist,
+            coord_ra_dec=self.online_ra_dec,
+        )
+        self.online_ra_dec = None
```

Comment:
Is the intention that `self.online_ra_dec` is only used on the first invocation of this method, and ignored on subsequent calls? If that's the case, then a conditional around `nsides_dict` would be more robust, though that specific conditional logic may be better placed _in_ `choose_pixels_to_reconstruct()`, ie, always pass `coord_ra_dec` (or a renamed version of the var, if needed).

Reply:
I implemented a `len(nside_progression) == 1` in `choose_pixels_to_reconstruct()`, which is `True` only at the first invocation of the method.
It is important to use `choose_pixels_to_reconstruct_around_coord()` only at the first invocation for two reasons:
1. If it is always called, and the successive nsides the scanner won't be refining the pixels, but instead just always scan again the same area in the sky with always finer pixels (and for the nsides 512 and 1024 they become a huge number);
2. It may happen (very rarely, I would say once every ~200-300 events) that the minimum is around the border of the area selected for the scan, in this way the scanner is not forced to keep scanning always in the same area but can explore nearby as well.